Apps/phsentinelone/sentinelone_connector.py

```diff
@@ -702,6 +702,56 @@ def _handle_get_cves(self, param):
             return action_result.get_status()
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _handle_get_device_control_events(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        ip_hostname = param['ip_hostname']
+        try:
+            ret_val = self._get_site_id(action_result)
+        except Exception:
+            return action_result.set_status(phantom.APP_ERROR, "Did not get proper response from the server")
+        self.save_progress('Agent query: {}'.format(ret_val))
+        if ret_val == '0':
+            return action_result.set_status(phantom.APP_ERROR, "Endpoint not found")
+        elif ret_val == '99':
+            return action_result.set_status(phantom.APP_ERROR, "More than one endpoint found")
+        else:
+            summary = action_result.update_summary({})
+            summary['ip_hostname'] = ip_hostname
+            summary['site_id'] = ret_val
+            header = self.HEADER
+            header["Authorization"] = "APIToken %s" % self.token
+            params = {"siteIds": ret_val}
```

Comment:
You have to filter and fetch the logs as per agent ID not site ID,
you have fetched all the events from site ID
Please fetch the device control events as per agent and not sites